app/src/androidTest/java/com/example/habitsmasher/MainActivityTest.java

```diff
@@ -790,4 +884,18 @@ private void assertTextOnScreen(String text) {
     private boolean isTextOnScreen(String text) {
         return _solo.waitForText(text, 1, 2000);
     }
+
+    private void assertCorrectDays(){
+        // I'm not entirely sure how to check the state of a button nor get the drawable without context, so
+        // my workaround checks the drawable of the buttons compared to each other, which change based on whether they're checked.
+        assertNotEquals(_solo.getView(R.id.tuesday_button).getBackground(),_solo.getView(R.id.monday_button).getBackground());
+        assertNotEquals(_solo.getView(R.id.wednesday_button).getBackground(), _solo.getView(R.id.thursday_button).getBackground());
+        assertNotEquals(_solo.getView(R.id.friday_button).getBackground(), _solo.getView(R.id.saturday_button).getBackground());
+        assertNotEquals(_solo.getView(R.id.sunday_button).getBackground(), _solo.getView(R.id.monday_button).getBackground());
+
+        // If you include these lines, you'll see that each button has the same background but different hex code, meaning
+        // Different states. This gives the difference between states.
+        //Log.d("Monday Background", _solo.getView(R.id.monday_button).getBackground().toString());
+        //Log.d("Tuesday Background", _solo.getView(R.id.tuesday_button).getBackground().toString());
+    }
 }
```

Comment:
This checks whether the correct background for the buttons is used when viewing the days of the week for a habit.